### PR TITLE
Make `_get_registry_uri_from_spark_session` return "databricks-uc" in case of Databricks Serverless client

### PR DESCRIPTION
**Pull request by WeichenXu123**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/WeichenXu123/mlflow/pull/11719?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11719/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11719
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Make `_get_registry_uri_from_spark_session` return "databricks-uc" in case of Databricks Serverless client

Context: https://databricks.slack.com/archives/C022KUWGVQS/p1712956996629289

### How is this PR tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [X] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by harupy** [comment]
@WeichenXu123 There is no plan to release 2.11.4 and 12.0 hasn't been released so No should be selected for `Should this PR be included in the next patch release?`.

File: mlflow/tracking/_model_registry/utils.py [hunk 1]
```diff
@@ -81,6 +82,12 @@ def _get_registry_uri_from_spark_session():
     session = _get_active_spark_session()
     if session is None:
         return None
+
+    dbr_version = get_databricks_runtime_version()
+    if dbr_version and dbr_version.startswith("client."):
+        # Connected to Serverless
+        return "databricks-uc"
```
Comment (harupy) [inline mlflow/tracking/_model_registry/utils.py:89]: Shouldn't this be used when `spark.mlflow.modelRegistryUri` is missing?

Comment (WeichenXu123) [reply mlflow/tracking/_model_registry/utils.py:89]: @BenWilson2 suggests to only provide default value in serverless: https://databricks.slack.com/archives/C022KUWGVQS/p1713194084260239?thread_ts=1712956996.629289&cid=C022KUWGVQS

Comment (harupy) [reply mlflow/tracking/_model_registry/utils.py:89]: got it.

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Pull request merged**